ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -171,7 +176,37 @@ private ExecuteResult execute(
       final Injector injector) {
     final PreparedStatement<?> prepared = executionContext.prepare(stmt);
     final ConfiguredStatement<?> configured =
-        ConfiguredStatement.of(prepared, overriddenProperties, ksqlConfig);
-    return executionContext.execute(injector.inject(configured));
+        injector.inject(ConfiguredStatement.of(prepared, overriddenProperties, ksqlConfig));
+
+    final CustomExecutor executor =
+        CustomExecutors.CUSTOM_EXECUTORS.getOrDefault(
+            configured.getStatement().getClass(),
+            executionContext::execute);
+
+    return executor.apply(configured);
+  }
+
+  @FunctionalInterface
+  private interface CustomExecutor extends Function<ConfiguredStatement<?>, ExecuteResult> { }
+
+  @SuppressWarnings("unchecked")
+  private static final class CustomExecutors {
+
+    private static final Map<Class<? extends Statement>, CustomExecutor> CUSTOM_EXECUTORS =
```

Comment:
Feels like these should be using `PropertyOverrider`.  Is there a reason they aren't?

Reply:
just future extensibility and to make it mirror the code in StandaloneExecutor and KsqlResource

Reply:
this was the behavior in KsqlContext in the past (it did not actually go through the validation). I can change this though